src/traceme.c

```diff
@@ -347,54 +447,128 @@ traceme_start_tracer(struct tracer_conf *tr_conf)
     DEBUG("Travis mode. Not starting tracing");
     return NULL;
 #else
+    int failing = 0;
```

Comment:
This should be a `bool` I think? [we are using C99 :)]

Reply:
Can do. Do you want all the functions that return an int for an error a bool too?

Reply:
I think we should not use bools at the API level, as this places an unnecessary constraint on the consuming C code. I've used a bool inside as you suggested though.

Reply:
We're never going to come across a compiler that doesn't at least understand C99 (an 18 year old standard remember!), so let's please use bools when they're the right datatype.

Reply:
so return false on error?

Reply:
I think it would be clearer that way.

Reply:
Hrm.

It looks like the libc crate doesn't have these. I was going to raise a bug, but it looks like the rust devs are well aware. 

https://github.com/rust-lang/rust/issues/14608
https://github.com/rust-lang/rfcs/issues/992

We can probably use one of the hacks show here:
https://github.com/rust-lang/rfcs/issues/992#issuecomment-264631983

But we need to be careful, because what these guys are implying is that the integer values under `true` and `false` are platform dependent! I didn't see that coming.

Reply:
I doubt there's any platform where they're not 1 and 0 respectively, but typical C not to actually define that :/

Reply:
OK, let's define them exactly as that and not rely on the rust bool being the same by chance.